File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/decorator/dataexport/CsvDataExportDecorator.java
// CsvDataExportDecorator.java

package com.bigboxcode.designpattern.decorator.dataexport;

public class CsvDataExportDecorator extends DataExportDecorator {
    public CsvDataExportDecorator(DataExport dataExporter) {
        super(dataExporter);
    }

    @Override
    public void processData() {
        dataExporter.processData();
        this.processCsv();
    }

    private void processCsv() {
        System.out.println("Processed data to CSV");
    }
}

File: design-patterns/php/decorator/dataexport/CsvDataExportDecorator.php
<?php
// CsvDataExportDecorator.php

namespace BigBoxCode\DesignPattern\Decorator\DataExport;

class CsvDataExportDecorator extends DataExportDecorator {
    public function __construct(DataExport $dataExporter) {
        parent::__construct($dataExporter);
    }

    public function processData(): void {
        $this->dataExporter->processData();
        $this->processCsv();
    }

    public function processCsv(): void {
        echo "Processed data to CSV\n";
    }
}